tests/test_external_roundtrip.rs

```diff
@@ -0,0 +1,106 @@
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate tempdir;
+
+use tempdir::TempDir;
+
+use std::path::Path;
+use std::process::Command;
+
+/// A macro used to simplify debugging.
+const DEBUG: bool = true;
+
+fn run(mut command: std::process::Command) -> Result<(), std::io::Error> {
+    let output = command.output()?;
+    if DEBUG || !output.status.success() {
+        debug!(target: "test", "Command: {command:?}\n** out:{out}\n** err:{err}",
+            command = command,
+            out =  String::from_utf8(output.stdout).unwrap_or_else(|_| "<cannot decode output>".to_string()),
+            err =  String::from_utf8(output.stderr).unwrap_or_else(|_| "<cannot decode output>".to_string())
+        );
+    }
+    assert!(output.status.success());
+    Ok(())
+}
+
+fn perform_test(root: &Path, prefix: &str, source: &str) {
+    let root = root.join(prefix);
+    let path_in_source = std::path::Path::new(source);
+    let path_tmp_source = root.join("sample.js");
+    let path_out_encoded_1 = root.join("encoded_1");
+    let path_in_encoded_1 = root.join("encoded_1").join("sample.binjs");
+    let path_out_encoded_2 = root.join("encoded_2");
+    let path_in_encoded_2 = root.join("encoded_2").join("sample.binjs");
+    let path_decoded = root.join("decoded.js");
+
+    let mut dir_builder = std::fs::DirBuilder::new();
+    dir_builder.recursive(true).create(root).unwrap();
+
+    // Copy source file to `path_tmp_source`, to normalize file name.
+    std::fs::copy(&path_in_source, &path_tmp_source).unwrap();
+
+    debug!(target: "test", "Encoding to encoded_1/sample.binjs");
+    let mut command_1 = Command::new("target/debug/binjs_encode");
```

Comment:
Hmm, I just realised that this depends on `binjs_encode` CLI app being already built, but this dependency/order is not enforced it in any way, which means that e.g. `cargo test` on clean checkout would fail.

Not sure if there's a good way to enforce it, but seems like something that should be at least explicitly documented (that you need to run `cargo build` to build all targets before testing)?

Reply:
I have added an explicit call to `cargo build` during the initialization of the test.